Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/BaseInteractiveElement.cs

```diff
@@ -0,0 +1,247 @@
+        {
+            return StateManager.AddCoreState(state);
+        }
+
+        /// <summary>
+        /// Creates and adds a new state to track given the new state name.
+        /// </summary>
+        /// <param name="stateName">The name of the state to add</param>
+        /// <returns>The new state added</returns>
+        public InteractionState AddNewState(string stateName)
+        {
+            return StateManager.AddNewState(stateName);
+        }
+
+        /// <summary>
+        /// Removes a Core Interaction State. The state will no longer be tracked if it is removed.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to remove</param>
+        public void RemoveCoreState(CoreInteractionState state)
+        {
+            StateManager.RemoveState(state);
+        }
+
+        /// <summary>
+        /// Removes a state. The state will no longer be tracked if it is removed.
+        /// </summary>
+        /// <param name="stateName">The name of the state to remove</param>
+        public void RemoveState(string stateName)
+        {
+            StateManager.RemoveState(stateName);
+        }
+
+        /// <summary>
+        /// Create and add a new state given the state name and the associated existing event configuration.
+        /// </summary>
+        /// <param name="stateName">The name of the state to create</param>
+        /// <param name="eventConfiguration">The existing event configuration for the new state</param>
+        /// <returns>The new state added</returns>
+        public void AddNewStateWithEventConfiguration(string stateName, BaseInteractionEventConfiguration eventConfiguration)
+        {
+            StateManager.AddNewStateWithEventConfiguration(stateName, eventConfiguration);
+        }
+
+        /// <summary>
+        /// Check if a state is currently being tracked.
+        /// </summary>
+        /// <param name="stateName">The name of the state to check</param>
+        /// <returns>True if the state is being tracked, false if the state is not being tracked</returns>
+        public bool IsStateTracking(string stateName)
+        {
+            return StateManager.IsStateTracking(stateName);
+        }
+
+        /// <summary>
+        /// Check if a state is currently active.
+        /// </summary>
+        /// <param name="stateName">The name of the state to check</param>
+        /// <returns>True if the state is active, false if the state is not active</returns>
+        public bool IsStateActive(string stateName)
+        {
+            return StateManager.IsStateActive(stateName);
+        }
+
+        /// <summary>
+        /// Check if a core state is currently active.
+        /// </summary>
+        /// <param name="coreState">The name of the core state to check</param>
+        /// <returns>True if the state is active, false if the state is not active</returns>
+        public bool IsStateActive(CoreInteractionState coreState)
```

Comment:
It's a little counter-intuitive to me that we have these two types -- CoreInteractionState and InteractionState. Why not just have everything be an InteractionState? What's the difference?

Reply:
The difference is that `CoreInteractionState` is an enum that lists the already defined states, right now this enum only contains `Focus` because that is the only state currently supported in `BaseInteractiveElement`.  While `InteractionState` is the type of the state itself, `InteractionState` stores the state name, value, active bool, and event configuration.  

Although I see how it is confusing to have an enum named `CoreInteractionState` while the type for every state is named an `InteractionState`.  I think a possible solution here is to remove the use of the `CoreInteractionState` enum as a parameter for some of the State Manager methods and only have the string state name as the main parameter to reduce confusion.

The original motivation behind the addition of the `CoreInteractionState` enum was for convivence and a way to communicate to the developer which states already have state setting logic defined.  For example, if a user wanted to get a core MRTK state (Focus, Touch, etc.), they would call `StateManager.GetState(CoreInteractionState.Focus)` instead of `StateManager.GetState("Focus")` (both of these options are valid now).  The only purpose of the enum is to tell the user which states are built into the component and do not require manual setting. 

If a developer were to create a new state, they would need to define when this new state is set to on/off.  A core state already has this logic defined and built into the component.